src/util/fetchUtils.js

```diff
@@ -27,35 +27,15 @@ export const parseQueryResponse = queries => ({ responses, error, message }) =>
 	};
 };
 
-/**
- * Wrapper around `fetch` to send an array of queries to the server. It ensures
- * that the request will have the required OAuth and CSRF credentials and constructs
- * the `fetch` call arguments based on the request method. It also records the
- * CSRF header value in a cookie for use as a CSRF header in future fetches.
- *
- * **IMPORTANT**: This function should _only_ be called from the browser. The
- * server should never need to call itself over HTTP
- *
- * @param {String} apiUrl the general-purpose endpoint for API calls to the
- *   application server
- * @param {Object} options {
- *     method: "get", "post", "delete", or "patch",
- *   }
- * @return {Promise} resolves with a `{queries, responses}` object
- */
-export const fetchQueries = (apiUrl, options) => (queries, meta) => {
-	if (
-		typeof window === 'undefined' &&  // not in browser
-		typeof test === 'undefined'  // not in testing env
-	) {
-		throw new Error('fetchQueries was called on server - cannot continue');
-	}
-	options.method = options.method || 'GET';
+export const getFetchArgs = (apiUrl, options, queries, meta) => {
 	const {
-		method,
 		headers={},
 	} = options;
 
+	const method = (queries[0].meta || {}).method ||  // allow query to set method
+		options.method.toLowerCase() ||  // fallback to options
+		'get';  // fallback to 'get'
```

Comment:
I think it would be a little weird just because those terms are used in lots of different contexts and the should be well-known - using a constant to enforce string case seems awkward, so I'll just coerce-lowercase whatever is supplied and be done with it.